File: Project/backend/app/models/preferenza_utente.py
from sqlmodel import SQLModel, Field

class PreferenzaUtente(SQLModel, table=True):
    __tablename__ = "preferenza_utente"

    id_utente: int = Field(primary_key=True, foreign_key="utente.id", nullable=False)
    id_categoria: int = Field(primary_key=True, foreign_key="categoria_poi.id", nullable=False)

class PreferenzaUtenteCreate(SQLModel):
    id_utente: int
    id_categoria: int

class PreferenzaUtentePublic(SQLModel):
    id_utente: int
    id_categoria: int

class PreferenzaUtenteUpdate(SQLModel):
    id_utente: int | None = Field(default=None)
    id_categoria: int | None = Field(default=None)

File: Project/ContextAware_app/app/src/main/java/com/example/myapplication/data/model/preferenza_utente.kt
package com.example.myapplication.data.model

import kotlinx.serialization.Serializable

@Serializable
data class PreferenzaUtenteCreate(
    val id_utente: Int,
    val id_categoria: Int
)

@Serializable
data class PreferenzaUtentePublic(
    val id_utente: Int,
    val id_categoria: Int
)

@Serializable
data class PreferenzaUtenteUpdate(
    val id_utente: Int? = null,
    val id_categoria: Int? = null
)